### PR TITLE
chore(deps): Upgrade USWDS, React USWDS, and Storybook

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1830

## Proposed changes

* Upgrades USWDS to 3.x and React USWDS to 5.x
* Upgrades Storybook to 7.x

---

## Reviewer notes

Two major changes in the USWDS upgrade affected us:
* USWDS renamed a lot of things
  * The module itself went from uswds --> `@uswds`
  * Instead of importing individual scss files from the module, we're pulling in packages
    * Compare changes in `styles/index.scss` for concrete examples
 * USWDS upgraded to the latest Sass best practices (using `@use` over `@import`)

Changes we made:
* Added `sassOptions` into our `next.config.js`, and passed in the path to the correct node_modules directory so our app can locate the appropriate uswds packages.
* Updated naming in script that copies fonts and img from uswds -> public vendor directory
* Updated all our scss files to use the modern `@forward` and `@use` syntax
  * `@import 'styles/uswdsDependencies'` → `@use styles/sfds/index.scss as *`
  * `@import 'styles/uswdsDependencies'` → `@use 'uswds-core' as *`
  * `as *` means we're adding everything from the imported stylesheet to the namespace of our module. We current don't have any naming collisions, so this is fine. But we can probably make better use of this feature in the future.

Changes for the future:
* There's opportunity to make better use of namespacing with `@use`
* Possible optimization over which packages we're pulling in (uswds-utilities, etc, instead of using uswds-core. Unclear if this is necessary for us, though.)

### Resources
* USWDS Migration docs - https://designsystem.digital.gov/documentation/migration/
* React USWDS docs - https://github.com/trussworks/react-uswds/blob/main/docs/styles_and_assets.md
* Sass docs - https://sass-lang.com/documentation/at-rules/use

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

File: src/styles/_uswdsDependencies.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
-@import 'node_modules/uswds/dist/scss/theme/uswds-theme-general';
-@import 'node_modules/uswds/dist/scss/theme/uswds-theme-typography';
-@import 'node_modules/uswds/dist/scss/theme/uswds-theme-spacing';
-@import 'node_modules/uswds/dist/scss/theme/uswds-theme-color';
-@import 'node_modules/uswds/dist/scss/theme/uswds-theme-utilities';
-@import './settings';
-@import 'node_modules/uswds/src/stylesheets/packages/_required.scss';
+// @forward 'theme/styles';
```
Comment (abbyoung) [inline src/styles/_uswdsDependencies.scss:1]: #TODO i think we can safely delete this file

File: src/components/Header/Header.test.tsx [hunk 2]
```diff
@@ -68,7 +68,7 @@ describe('Header component', () => {
 
     expect(nav).toHaveClass('is-visible')
 
-    await user.click(screen.getByRole('button', { name: 'close' }))
+    await user.click(screen.getByRole('button', { name: 'Close' }))
```
Comment (abbyoung) [inline src/components/Header/Header.test.tsx:71]: I'm not sure why this test decided to start failing for me, but it seemed to be a case-sensitivity error

**Review by shkeating — COMMENTED** [review]
Things are looking good, only thing I noticed is the icons for some components (carats and the cog for widget settings) are showing up as boxes. 
Screenshot_20230630-100150
Screenshot_20230630-100002
Screenshot_20230630-095934

**Comment by gidjin** [comment]
Good news is I see this issue, but only in storybook, as evidenced by the header from the app running. I think that means I am missing some portion of the css setup for storybook. I'll take a look

SCR-20230630-bzj

**Comment by jcbcapps** [comment]
Two things I've noticed:
The announcement has lost padding/margin on My Space page. This is not happening on the `/news` page though, which is kind of strange, so probably something with `DefaultLayout`?
announcement


On `/news` the button on the bottom of the page has the dark theme styles while the light theme is applied.
button

File: src/styles/_uswdsDependencies.scss [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+// DEPRECATED AFTER USWDS 3.0 UPGRADE
+// #TODO REMOVE AFTER UPGRADE COMPLETE
+
+// @forward 'theme/styles';
+// @forward 'node_modules/uswds/dist/scss/theme/uswds-theme-typography';
+// @forward 'node_modules/uswds/dist/scss/theme/uswds-theme-spacing';
+// @forward 'node_modules/uswds/dist/scss/theme/uswds-theme-color';
+// @forward 'node_modules/uswds/dist/scss/theme/uswds-theme-utilities';
+// @forward './settings';
+// @forward 'node_modules/uswds/src/stylesheets/packages/_required.scss';
```
Comment (jcbcapps) [inline src/styles/_uswdsDependencies.scss:10]: Should we go ahead and remove this file?

**Review by jcbcapps — APPROVED** [review]
I left a few comments/screenshots of things that I think need to be addressed, but nothing too crazy. Great work!

**Comment by shkeating** [comment]
aw JEEZ i just realized the dark mode toggle is missing in storybook 🫠

File: src/styles/_uswdsDependencies.scss (hunk 3 above)
Comment (gidjin) [reply src/styles/_uswdsDependencies.scss:10]: Yeah I think removing it makes sense

File: src/styles/pages/home.module.scss [hunk 4]
```diff
@@ -1,5 +1,3 @@
-@import 'styles/uswdsDependencies';
-
 .home {
   section + section {
     margin-top: units('205');
```
[the remark this answers is not in the document]
Comment (gidjin) [reply src/styles/pages/home.module.scss:3]: I'm not sure but I see them in the other pages so I will try adding them.

File: src/styles/pages/annoucements.module.scss [hunk 5]
```diff
@@ -1,6 +1,3 @@
-@import 'styles/uswdsDependencies';
-@import 'styles/sfds/sfdsDependencies';
-
 .pageTitle {
   margin-top: units(6);
   margin-bottom: units(4);
```
[the remark this answers is not in the document]
Comment (gidjin) [reply src/styles/pages/annoucements.module.scss:3]: I'm not sure but I see them in the other pages so I will try adding them. I think it's a possibility that it messed up the margins.

[the remark this answers is not in the document]
Comment (gidjin) [reply src/styles/pages/annoucements.module.scss:3]: looks like you were right! Thanks for the suggestion and 🦅 👁️ catch

SCR-20230703-nd2

**Comment by shkeating** [comment]
image

the `-webkit-` prefix is not getting added to the icons in storybook, but it is in the app. we need to figure out whats going on with prefixing in the storybook build

File: .storybook/preview.js [hunk 6]
```diff
@@ -5,11 +5,12 @@ import { MockedProvider } from '@apollo/client/testing'
...
-import 'uswds/dist/css/uswds.css'
-import '../src/initIcons'
-import '../src/styles/index.scss'
+import 'styles/index.scss'
+// #TODO these imports seem unneccessary can we delete them?
 // SFDS
-import '../src/styles/sfds/index.scss'
+//import 'styles/sfds/index.scss'
+// import '../src/styles/sfds/index.scss'
+//import 'initIcons'
```
Comment (gidjin) [inline .storybook/preview.js:13]: I'm going to remove these commented out lines, I don't think we need them. Objections?

**Comment by shkeating** [comment]
it seems like all uswds icons are restored, but fontawesome ones are still missing:

- arrows in latest annoucments carousel
- drag handle icons
- widget settings cog icon on all widgets
- news carousel arrow icons
- edit and delete icons on bookmarks

File: .storybook/preview.js (hunk 6 above)
Comment (abbyoung) [reply .storybook/preview.js:13]: i was going to leave them until every issue was resolved, just to make sure it was okay to delete

Comment (abbyoung) [reply .storybook/preview.js]: update: we needed initIcons!

**Comment by shkeating** [comment]
looks good!! 🥳

**Review by shkeating — APPROVED** [review]

File: src/styles/index.scss [hunk 7]
```diff
@@ -1,14 +1,14 @@
 // ReactUSWDS CSS modules
-@import '@trussworks/react-uswds/lib/index.css';
+@forward 'node_modules/@trussworks/react-uswds/lib/index.css';
+@use 'node_modules/@trussworks/react-uswds/lib/index.css'; // #TODO do we need to use?
```
Comment (gidjin) [inline src/styles/index.scss:3]: question: What's the verdict here do we need this line? Probably yes?

File: src/components/SelectableCollection/SelectableCollection.module.scss [hunk 8]
```diff
@@ -1,6 +1,5 @@
-@import 'styles/uswdsDependencies';
-@import 'styles/sfds/sfdsDependencies';
-@import 'styles/deps';
```
Comment (gidjin) [inline src/components/SelectableCollection/SelectableCollection.module.scss:3]: Do we need to add `@use 'styles/deps'` here?

**Review by gidjin — APPROVED** [review]

File: src/components/SelectableCollection/SelectableCollection.module.scss (hunk 8 above)
Comment (abbyoung) [reply src/components/SelectableCollection/SelectableCollection.module.scss:3]: `styles/deps` contains button reset, which isn't used in this file, so we don't need it 👍

File: src/styles/index.scss (hunk 7 above)
Comment (gidjin) [reply src/styles/index.scss]: Removed and everything seems fine.

**Pull request merged**